assets/js/components/DataBlock.js

```diff
@@ -36,193 +36,164 @@ import SourceLink from './SourceLink';
 import GatheringDataNotice, { NOTICE_STYLE } from './GatheringDataNotice';
 import { numFmt } from '../util';
 
-class DataBlock extends Component {
-	constructor( props ) {
-		super( props );
-
-		this.handleClick = this.handleClick.bind( this );
-		this.handleKeyPress = this.handleKeyPress.bind( this );
-	}
-
-	handleClick() {
-		const { stat, handleStatSelection } = this.props;
-		handleStatSelection( stat );
-	}
+const DataBlock = ( {
+	stat,
+	className = '',
+	title = '',
+	datapoint,
+	datapointUnit = undefined,
+	change,
+	changeDataUnit = '',
+	context = 'default',
+	period = '',
+	selected = false,
+	source,
+	sparkline,
+	handleStatSelection,
+	invertChangeColor = false,
+	gatheringData = false,
+	gatheringDataNoticeStyle = NOTICE_STYLE.DEFAULT,
+} ) => {
+	const handleClick = () => {
```

Comment:
Please, use `useCallback` for all callbacks to make it consistent with other components.